app/models.py

```diff
@@ -778,14 +789,47 @@ def submitted(cls, session: Session) -> "Query[Self]":
         )
 
     @classmethod
-    def submitted_to_lender(cls, session: Session, lender_id: int | None) -> "Query[Self]":
-        """
-        :return: A query for applications :meth:`~app.models.Application.submitted` to a specific lender.
-        """
-        return cls.submitted(session).filter(
-            Application.lender_id == lender_id,
-            col(Application.lender_id).isnot(None),
+    def submitted_search(
+        cls,
+        session: Session,
+        sort_field: str,
+        sort_order: str,
+        lender_id: int | None = None,
+        search_value: str | None = None,
+    ):
+        applications_query = (
+            cls.submitted(session)
+            .join(Award)
+            .join(Borrower)
+            .join(CreditProduct)
+            .join(Lender)
+            .options(
+                joinedload(Application.award),
+                joinedload(Application.borrower),
+                joinedload(Application.borrower_documents),
+                joinedload(Application.credit_product),
+                joinedload(Application.lender),
+            )
+            .order_by(get_order_by(sort_field, sort_order, model=Application))
         )
+        if search_value:
+            applications_query = applications_query.filter(
+                or_(
+                    Application.primary_email == search_value,
+                    col(Borrower.legal_name).ilike(f"%{search_value}%"),
+                    col(Borrower.legal_identifier).ilike(f"%{search_value}%"),
+                    col(Award.buyer_name).ilike(f"%{search_value}%"),
+                    col(Lender.name).ilike(f"{search_value}%"),
```

Comment:
Is Lender.name intended to be a prefix search, or is there a missing `%`?

Reply:
Also, shouldn't the lender filter only apply if there is both a search_value and `not lender_id`? Otherwise, if the search text happens to match the lender, then the lender will get all their applications in response to the search, which is not expected.

Reply:
> then the lender will get all their applications in response to the search

I don't think so, as the lender_id is an AND condition. But this one was for our (mine) internal use and wasn't requested as part of the issue so thinking about it again I will just remove it for now.

Reply:
Hmm, it looked like they were all `or_` together - but no matter now.